Prose/ProseLib/Sources/AuthenticationFeature/AuthenticationReducer.swift

```diff
@@ -62,30 +95,64 @@ public let authenticationReducer = Reducer<
     AuthenticationState,
     AuthenticationAction,
     AuthenticationEnvironment
-> { state, action, _ in
+> { state, action, environment in
+    struct CancelId: Hashable {}
```

Comment:
No need to act on that, just as a note:

Maybe you already have something like that, however I've came to love this extension…

```swift
public extension Effect {
  // Source: https://forums.swift.org/t/ifletstore-and-effect-cancellation-on-view-disappear/38272/23
  static func cancel<T>(token: T.Type) -> Effect where T: CaseIterable, T: Hashable {
    .merge(token.allCases.map(Effect.cancel(id:)))
  }
}
```

You'd then define "effect tokens", e.g.

```swift
enum InboxEffectToken: CaseIterable, Hashable {
  case upgradeSubscription
  case requestTrackingAuthorization
}
```

and cancel them all at once…

```swift
case .onDisappear:
  .cancel(token: InboxEffectToken.self)
```

Reply:
😯 That's handy 👍🏻 Thank you 🙏🏻